### PR TITLE
Coq 8.15.2

### PR DESCRIPTION
**Pull request by kazk**
Using the same versions as Coq Platform 2022.04.1 for Coq 8.15.

https://github.com/coq/platform/blob/main/package_picks/package-pick-8.15%7E2022.04.sh

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,80 +1,59 @@
...
-      xz \
-    ;
-
-# Disable sandboxing
-RUN set -ex; \
-    echo 'wrap-build-commands: []' > ~/.opamrc; \
-    echo 'wrap-install-commands: []' >> ~/.opamrc; \
-    echo 'wrap-remove-commands: []' >> ~/.opamrc; \
-    echo 'required-tools: []' >> ~/.opamrc;
+FROM alpine:3.16
```
Comment (kazk) [inline Dockerfile:1]: We can switch to `ubuntu` if we run into any issues. Alpine is just more convenient to remove build deps. I'll accept a PR to migrate if the final image size is comparable.

I doubt a possibility of slight version shifts in system dependencies when rebuilding the image will be a problem for Coq kata, but we'll see.

File: Dockerfile [hunk 2]
```diff
@@ -1,80 +1,59 @@
...
+        opam \
+        tar \
+    ; \
+    opam init -y --disable-sandboxing; \
+    opam repo add coq-released http://coq.inria.fr/opam/released; \
+    opam install -j 4 \
+    # Let opam install system packages
+        --confirm-level=unsafe-yes \
+    # Use the same versions as Coq Platform so users can easily get the same versions.
+    # https://github.com/coq/platform/blob/main/package_picks/package-pick-8.15%7E2022.04.sh
```
Comment (kazk) [inline Dockerfile:38]: Package versions must match Coq Platform package set.

**Comment by Zimmi48** [comment]
@kazk It would be really great though if the package set used could be extended beyond just Equations and mathcomp-ssreflect. The Coq Platform is a standard in the Coq ecosystem, as it contains many small but very convenient packages that Coq users may want to use or may learn about when asking questions in Coq's Zulip forum. Not providing them would be a bit like not providing mathlib in Lean (you are really disarmed without them). If you provide a size limit, the Coq Platform team may be able to come up with a list of recommended small but useful Coq Platform packages that fits in this limit.

cc Coq Platform team members @MSoegtropIMC @palmskog @gares

**Comment by kazk** [comment]
Yeah, we can extend as needed, like other languages. Equations and mathcomp-ssreflect are installed because we already have them in 8.12.

Anything we add should be already useful on Codewars, or is required to author a new kata. Might be easier to start adding a few as needed instead of coming up with a list.

> If you provide a size limit

The image is currently 1.14 GB, and I'd like to keep it under 1.5 GB if possible.
Most of our images are under 1 GB. Agda is 278 MB.

**Comment by palmskog** [comment]
@kazk  the mathlib version for Lean you already have on Code Wars already contains around 350k lines of Lean code. In contrast, the Coq libraries available (Stdlib + mathcomp-ssreflect) are around 170k lines of Coq code. Can you explain how your minimization requirements apply across both Coq and Lean?

**Comment by kazk** [comment]
The number of lines of code is irrelevant.
I actually meant to mention Lean and not Agda because @Zimmi48 mentioned it. Our Lean image is 844 MB.

Like I wrote above, most of our images are under 1 GB, and I'd prefer to keep Coq under 1.5 GB if possible.

**Pull request merged**